rskj-core/src/test/java/co/rsk/peg/federation/FederationChangeIT.java

```diff
@@ -0,0 +1,570 @@
+import java.util.List;
+import java.util.Optional;
+import org.bouncycastle.util.encoders.Hex;
+import org.ethereum.TestUtils;
+import org.ethereum.config.Constants;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ActivationConfigsForTest;
+import org.ethereum.config.blockchain.upgrades.ConsensusRule;
+import org.ethereum.core.*;
+import org.ethereum.crypto.ECKey;
+import org.ethereum.vm.PrecompiledContracts;
+import org.junit.jupiter.api.Test;
+
+class FederationChangeIT {
+
+    private static final BridgeConstants BRIDGE_CONSTANTS = BridgeMainNetConstants.getInstance();
+    private static final List<FederationMember> ORIGINAL_FEDERATION_MEMBERS = FederationTestUtils.getFederationMembers(9);
+    private static final List<FederationMember> NEW_FEDERATION_MEMBERS = FederationTestUtils.getFederationMembers(9);
+    private static final SignatureCache SIGNATURE_CACHE = new BlockTxSignatureCache(new ReceivedTxSignatureCache());
+    private static final Transaction UPDATE_COLLECTIONS_TX = buildUpdateCollectionsTx();
+    private static final Transaction FIRST_AUTHORIZED_TX = TransactionUtils.getTransactionFromCaller(SIGNATURE_CACHE, FederationChangeCaller.FIRST_AUTHORIZED.getRskAddress());
+    private static final Transaction SECOND_AUTHORIZED_TX = TransactionUtils.getTransactionFromCaller(SIGNATURE_CACHE, FederationChangeCaller.SECOND_AUTHORIZED.getRskAddress());
+    private static final ActivationConfig.ForBlock ACTIVATIONS = ActivationConfigsForTest.all().forBlock(0);
+
+    private Repository repository;
+    private BridgeStorageProvider bridgeStorageProvider;
+    private BtcBlockStoreWithCache.Factory btcBlockStoreFactory;
+    private BtcBlockStore btcBlockStore;
+    private BridgeEventLogger bridgeEventLogger;
+    private FeePerKbSupport feePerKbSupport;
+    private Block currentBlock;
+    private StorageAccessor bridgeStorageAccessor;
+    private FederationStorageProvider federationStorageProvider;
+    private FederationSupportImpl federationSupport;
+    private LockingCapSupport lockingCapSupport;
+    private BridgeSupport bridgeSupport;
+
+    @Test
+    void whenAllActivationsArePresentAndFederationChanges_shouldSuccesfullyChangeFederation() throws Exception {
+        // Arrange
+   
+        setUp();
+        // Create a default original federation using the list of UTXOs
+        var originalFederation = createOriginalFederation(ORIGINAL_FEDERATION_MEMBERS);
+        var originalUTXOs = federationStorageProvider.getNewFederationBtcUTXOs(BRIDGE_CONSTANTS.getBtcParams(), ACTIVATIONS);
+       
+        // Act & Assert
+   
+        // Create pending federation using the new federation keys
+        voteToCreateEmptyPendingFederation();
+        voteToAddFederatorPublicKeysToPendingFederation(NEW_FEDERATION_MEMBERS);
+
+        var pendingFederation = federationStorageProvider.getPendingFederation();
+        assertNotNull(pendingFederation);
```

Comment:
assert size and members?

Reply:
Added new method that includes these validations, `assertPendingFederationIsBuiltAsExpected()`